build-script.py

```diff
@@ -345,24 +298,23 @@ def run_lit_tests(swift_build_exec, build_dir, parser_header_dir, parser_lib_dir
 
 ## XCTest based tests
 
-def run_xctests(swift_test_exec, build_dir, parser_header_dir, parser_lib_dir,
-                release, swiftc_exec, verbose):
+def run_xctests(toolchain, build_dir, release, verbose):
     print('** Running XCTests **')
-    swiftpm_call = get_swiftpm_invocation(spm_exec=swift_test_exec,
-                                          build_dir=build_dir,
-                                          parser_header_dir=parser_header_dir,
-                                          parser_lib_dir=parser_lib_dir,
+
+    # Testing builds with testability enabled. Don't overwrite the
+    # build that we have just created and which shall be reused by
+    # dependent projects. Instead, build in a new directory.
+    test_build_dir = os.path.join(realpath(build_dir), 'test-build')
```

Comment:
I don't think enabling testability is important enough to force a separate build. For sourcekit-lsp we chose to test the normal release build and avoid enabling testability \cc @benlangmuir

Reply:
Another benefit of not using `-enable-testing` is you can test exactly the same configuration that goes in the toolchain.

Reply:
Ah, I found the issue, why it was rebuilding now. I was missing the `SWIFT_SYNTAX_CI_ENVIRONMENT` environment variable when testing and thus building with a different build configuration. Fixed that now.